### PR TITLE
Add overrideGUID option to PoolOutputModule

### PR DESCRIPTION
**Pull request by makortel**
#### PR description:

This PR adds `overrideGUID` parameter to `PoolOutputModule`. The intended use case is to ensure consistency of the GUID in the file name and in the file for the manual recovery at Tier0, requested in https://github.com/cms-sw/cmssw/issues/37742#issuecomment-1116227468.

#### PR validation:

Unit tests run.

### CODE REVIEW

**Comment by makortel** [comment]
@cmsbuild, please test

**Comment by makortel** [comment]
@Dr15Jones, @dan131riley please review

File: FWCore/Utilities/interface/GlobalIdentifier.h [hunk 1]
```diff
@@ -4,6 +4,8 @@
 #include <string>
 namespace edm {
   std::string createGlobalIdentifier(bool binary = false);
-}
+
+  bool isValidGlobalIdentifier(std::string const& guid);
```
Comment (Dr15Jones) [inline FWCore/Utilities/interface/GlobalIdentifier.h:8]: When can we have a non-valid one? Maybe a comment here?

OK, now I see this is testing that the string matches the required format.

File: IOPool/Output/src/RootOutputFile.cc [hunk 2]
```diff
@@ -196,7 +197,15 @@ namespace edm {
     parentageTree_ = RootOutputTree::makeTTree(filePtr_.get(), poolNames::parentageTreeName(), 0);
     parameterSetsTree_ = RootOutputTree::makeTTree(filePtr_.get(), poolNames::parameterSetsTreeName(), 0);
 
-    fid_ = FileID(createGlobalIdentifier());
+    if (overrideGUID.empty()) {
```
Comment (Dr15Jones) [inline IOPool/Output/src/RootOutputFile.cc:200]: Doesn't PoolOutputModule support writing multiple files (in the condition a file gets too large)? In that case, how does this override interact with it?

**Review by Dr15Jones — CHANGES_REQUESTED** [review]

File: IOPool/Output/src/RootOutputFile.cc (hunk 2 above)
Comment (makortel) [reply IOPool/Output/src/RootOutputFile.cc:200]: I wondered about that case too. With this PR both files would get the same GUID (from the configuration). That doesn't sound very good.

I can think of some alternatives:
* first file gets the GUID from configuration, and subsequent files get a generated GUID
* prevent multiple output files per PoolOutputModule instance (e.g. throw an exception if second file is opened)

But those don't sound very attractive either. Maybe subsequent files getting a generated GUID would be the least bad option? Or would you have more ideas?

Comment (makortel) [reply IOPool/Output/src/RootOutputFile.cc:200]: The last commit implements the 

> * first file gets the GUID from configuration, and subsequent files get a generated GUID

approach

**Comment by makortel** [comment]
@cmsbuild, please test

**Comment by makortel** [comment]
The unit test fails also in IBs (#37888).

**Comment by makortel** [comment]
+1

**Comment by qliphy** [comment]
+1

**Comment by qliphy** [comment]
merge

**Pull request merged**